pylti/tests/test_flask.py

```diff
@@ -382,84 +391,88 @@ def test_access_to_oauth_resource_post_grade_fail(self):
     def test_access_to_oauth_resource_post_grade_fix_url(self):
         """
-        Make sure URL remap works for edX vagrant stack
+        Make sure URL remap works for edX vagrant stack.
         """
         uri = 'https://localhost:8000/dev_stack'
 
-        def request_callback(request, cburi, headers):
-            return 200, headers, self.expected_response
-
-        httpretty.register_uri(httpretty.POST, uri, body=request_callback)
+        httpretty.register_uri(httpretty.POST, uri, body=self.request_callback)
 
         url = 'http://localhost/initial?'
-        new_url = self.generate_launch_request(self.consumers, url,
-                                               lit_outcome_service_url=uri)
+        new_url = TestFlask.generate_launch_request(
+            self.consumers, url, lit_outcome_service_url=uri
+        )
         ret = self.app.get(new_url)
-        self.assertFalse(self.has_exception())
+        self.assertFalse(TestFlask.has_exception())
 
         ret = self.app.get("/post_grade/1.0")
-        self.assertFalse(self.has_exception())
+        self.assertFalse(TestFlask.has_exception())
         self.assertEqual(ret.data, "grade=True")
 
         ret = self.app.get("/post_grade/2.0")
-        self.assertFalse(self.has_exception())
+        self.assertFalse(TestFlask.has_exception())
         self.assertEqual(ret.data, "grade=False")
 
     @httpretty.activate
     def test_access_to_oauth_resource_post_grade2(self):
         """
-        Check post_grade edX LTI2 functionality
+        Check post_grade edX LTI2 functionality.
         """
-
         uri = (u'https://example.edu/courses/MITx/ODL_ENG/2014_T1/xblock/'
                u'i4x:;_;_MITx;_ODL_ENG;_lti;'
                u'_94173d3e79d145fd8ec2e83f15836ac8/handler_noauth'
                u'/lti_2_0_result_rest_handler/user/'
                u'008437924c9852377e8994829aaac7a1')
 
-        def request_callback(request, cburi, headers):
-            return 200, headers, self.expected_response
-
-        httpretty.register_uri(httpretty.PUT, uri, body=request_callback)
+        httpretty.register_uri(httpretty.PUT, uri, body=self.request_callback)
 
         consumers = self.consumers
         url = 'http://localhost/initial?'
-        new_url = self.generate_launch_request(consumers, url)
+        new_url = TestFlask.generate_launch_request(consumers, url)
 
         ret = self.app.get(new_url)
-        self.assertFalse(self.has_exception())
+        self.assertFalse(TestFlask.has_exception())
 
         ret = self.app.get("/post_grade2/1.0")
-        self.assertFalse(self.has_exception())
+        self.assertFalse(TestFlask.has_exception())
         self.assertEqual(ret.data, "grade=True")
 
         ret = self.app.get("/post_grade2/2.0")
-        self.assertFalse(self.has_exception())
+        self.assertFalse(TestFlask.has_exception())
         self.assertEqual(ret.data, "grade=False")
 
+    def request_callback(self, request, cburi, headers):
```

Comment:
This is a mock callback. I don't think making it an instance method would work or is a good idea. Please move it outside the class definition.